state.go

```diff
@@ -129,8 +137,19 @@ type StateRunner interface {
 	// Obtain runner's state
 	GetState() State
 	// TODO generic state run action
+	RunState(ctx *StateContext) (State, bool)
```

Comment:
I don't get the idea. How is `RunState()` different from `Handle()` and why we need both?